src/lib.rs

```diff
@@ -289,3 +290,49 @@ pub use self::decode_all::DecodeAll;
 pub use self::depth_limit::DecodeLimit;
 pub use self::encode_append::EncodeAppend;
 pub use self::encode_like::{EncodeLike, Ref};
+pub use max_encoded_len::MaxEncodedLen;
+/// Derive macro for [`MaxEncodedLen`][max_encoded_len::MaxEncodedLen].
+///
+/// # Examples
+///
+/// ```
+/// # use parity_scale_codec::{Encode, MaxEncodedLen};
+/// #[derive(Encode, MaxEncodedLen)]
+/// struct Example;
+/// ```
+///
+/// ```
+/// # use parity_scale_codec::{Encode, MaxEncodedLen};
+/// #[derive(Encode, MaxEncodedLen)]
+/// struct TupleStruct(u8, u32);
+///
+/// assert_eq!(TupleStruct::max_encoded_len(), u8::max_encoded_len() + u32::max_encoded_len());
+/// ```
+///
+/// ```
+/// # use parity_scale_codec::{Encode, MaxEncodedLen};
+/// #[derive(Encode, MaxEncodedLen)]
+/// enum GenericEnum<T> {
+/// 	A,
+/// 	B(T),
+/// }
+///
+/// assert_eq!(GenericEnum::<u8>::max_encoded_len(), u8::max_encoded_len() + u8::max_encoded_len());
+/// assert_eq!(GenericEnum::<u128>::max_encoded_len(), u8::max_encoded_len() + u128::max_encoded_len());
+/// ```
+///
+/// # Within other macros
+///
+/// Sometimes the `MaxEncodedLen` trait and macro are used within another macro, and it can't be
+/// guaranteed that the `max_encoded_len` module is available at the call site. In that case, the
+/// macro should reexport the `max_encoded_len` module and specify the path to the reexport:
+///
+/// ```ignore
+/// pub use parity_scale_codec::max_encoded_len;
+///
+/// #[derive(Encode, MaxEncodedLen)]
+/// #[max_encoded_len_mod($crate::max_encoded_len)]
```

Comment:
I would really like to go the same way as serde https://serde.rs/container-attrs.html#crate

```
#[codec(crate = )]
```

Reply:
Maybe we can use the existing [`wrap_with_dummy_const`](https://github.com/paritytech/parity-scale-codec/blob/c5d9754374e708e899efa465de7e707d6305e277/derive/src/lib.rs#L56) trick and for now work on properly incorporating a
```
#[codec(crate = )]
```
like attribute in a subsequent (pre-)release? Ideally, we should have this new attribute work on both `{Encode, Decode}` and `MaxEncodedLen` but that's going to take a bit of work IIUC and I'd like to get the ball rolling on fixing the Substrate release with what we have now.